src/preamble.js

```diff
@@ -992,11 +992,21 @@ function createWasm(env) {
 #endif
 
 #if LOAD_SOURCE_MAP
+#if USE_PTHREADS
+  if (!ENVIRONMENT_IS_PTHREAD) {
```

Comment:
this pattern may be worth simplifying. how about
```
  {{{ runOnMainThread(`addRunDependency('source-map');`) }}}
```
and that function would be defined in `src/parseTools.js`?

Could be even better to avoid a string there, if we had something like
```
#mainThread
addRunDependency('source-map');
#end
```
but that would requires preprocessor changes, not sure it's worth it?

Reply:
I think the first option is good enough for all the use cases in `createWasm`. This should simplify the code quite a bit.